src/components/RegionSelect/RegionSelect.tsx

```diff
@@ -1,68 +1,130 @@
+  },
+  { groupKey: 'coral_reef_regions', groupName: 'Coral reef regions', label: 'Ningaloo Reef' },
 ]
 /** End of mock data */
 
 export default function RegionSelect() {
   const { t } = useTranslation()
-  const [selectedValue, setSelectedValue] = useState<CountryOption | null>(DEFAULT_OPTION)
+  const [selectedValue, setSelectedValue] = useState<RegionOption | null>(defaultOption)
+  const noCountriesMatchText = t('no_countries_match')
+  const noRegionsMatchText = t('no_regions_match')
 
-  const sortedOptions = countryOptions.sort((a, b) => {
-    if (a.group !== b.group) {
-      return a.group.localeCompare(b.group)
-    }
-    return a.label.localeCompare(b.label)
-  })
+  const sortedOptions = useMemo(() => {
+    return [...regionOptions].sort((a, b) => {
+      if (a.groupKey !== b.groupKey) {
+        const aIndex = groupOrders.indexOf(a.groupKey)
+        const bIndex = groupOrders.indexOf(b.groupKey)
+        return aIndex - bIndex
+      }
+      return a.label.localeCompare(b.label)
+    })
+  }, [])
 
-  const handleChange = (_, newValue: CountryOption | null) => {
-    setSelectedValue(newValue || DEFAULT_OPTION)
-  }
+  const muiFilterOptions = createFilterOptions<RegionOption>({ ignoreAccents: true, trim: true })
 
-  const isOptionEqual = (option: CountryOption, value: CountryOption) => {
-    return option.label === value.label && option.group === value.group
+  const createEmptyFilterOptions = useCallback((): RegionOption[] => {
+    const noDataGroups = [
+      {
+        groupKey: 'countries_with_coral',
+        groupName: 'Countries with Coral Reefs',
+        label: noCountriesMatchText,
+      },
+      {
+        groupKey: 'coral_reef_regions',
+        groupName: 'Coral reef regions',
+        label: noRegionsMatchText,
+      },
+    ]
+
+    return noDataGroups as RegionOption[]
+  }, [noCountriesMatchText, noRegionsMatchText])
+
+  const handleChange = (_: unknown, newValue: RegionOption | null) => {
+    setSelectedValue(newValue || defaultOption)
   }
 
   return (
     <div className={styles['RegionSelect-root']}>
-      <Autocomplete
+      <Autocomplete<RegionOption>
         size="small"
         options={sortedOptions}
-        groupBy={(option) => option.group}
+        groupBy={(option) => t(option.groupKey)}
         getOptionLabel={(option) => option.label}
+        getOptionDisabled={(option) =>
+          option.label === noCountriesMatchText || option.label === noRegionsMatchText
+        }
         aria-label={t('select_region')}
         value={selectedValue}
         onChange={handleChange}
-        isOptionEqualToValue={isOptionEqual}
+        isOptionEqualToValue={(option, value) =>
+          option.label === value.label && option.groupKey === value.groupKey
+        }
+        noOptionsText=""
+        filterOptions={(options, state) => {
+          const filtered = muiFilterOptions(options, state)
+          return filtered.length ? filtered : createEmptyFilterOptions()
```

Comment:
Non-blocking suggestion: 
```suggestion
         return muiFilterOptions(options, state) ?? createEmptyFilterOptions
```

Reply:
this is not valid, muiFilterOptions(options, state)  returns `[]`.